wear/src/main/java/io/homeassistant/companion/android/complications/views/ComplicationConfigMainView.kt

```diff
@@ -0,0 +1,135 @@
+import androidx.wear.compose.navigation.composable
+import androidx.wear.compose.navigation.rememberSwipeDismissableNavController
+import com.mikepenz.iconics.compose.Image
+import com.mikepenz.iconics.typeface.library.community.material.CommunityMaterial
+import io.homeassistant.companion.android.HomeAssistantApplication
+import io.homeassistant.companion.android.common.R
+import io.homeassistant.companion.android.complications.ComplicationConfigViewModel
+import io.homeassistant.companion.android.data.SimplifiedEntity
+import io.homeassistant.companion.android.home.views.ChooseEntityView
+import io.homeassistant.companion.android.home.views.ListHeader
+import io.homeassistant.companion.android.home.views.ThemeLazyColumn
+import io.homeassistant.companion.android.theme.WearAppTheme
+import io.homeassistant.companion.android.theme.wearColorPalette
+import io.homeassistant.companion.android.util.getIcon
+
+private const val SCREEN_MAIN = "main"
+private const val SCREEN_CHOOSE_ENTITY = "choose_entity"
+
+@OptIn(ExperimentalWearMaterialApi::class)
+@Composable
+fun LoadConfigView(
+    complicationConfigViewModel: ComplicationConfigViewModel,
+    onAcceptClicked: () -> Unit
+) {
+    WearAppTheme {
+        val swipeDismissableNavController = rememberSwipeDismissableNavController()
+        SwipeDismissableNavHost(
+            navController = swipeDismissableNavController,
+            startDestination = SCREEN_MAIN
+        ) {
+            composable(SCREEN_MAIN) {
+                MainConfigView(
+                    entity = complicationConfigViewModel.selectedEntity.value,
+                    hasSelected = complicationConfigViewModel.hasSelected.value,
+                    loaded = complicationConfigViewModel.loadingState.value == ComplicationConfigViewModel.LoadingState.READY,
+                    error = complicationConfigViewModel.loadingState.value == ComplicationConfigViewModel.LoadingState.ERROR,
+                    onChooseEntityClicked = {
+                        swipeDismissableNavController.navigate(SCREEN_CHOOSE_ENTITY)
+                    },
+                    onAcceptClicked = onAcceptClicked
+                )
+            }
+            composable(SCREEN_CHOOSE_ENTITY) {
+                val app = complicationConfigViewModel.getApplication<HomeAssistantApplication>()
+                ChooseEntityView(
+                    app.applicationContext,
+                    complicationConfigViewModel.entitiesByDomainOrder,
+                    complicationConfigViewModel.entitiesByDomain,
+                    {}, { entity ->
+                    complicationConfigViewModel.setEntity(entity)
+                    swipeDismissableNavController.navigateUp()
+                },
+                    false
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun MainConfigView(
+    entity: SimplifiedEntity,
+    hasSelected: Boolean,
+    loaded: Boolean,
+    error: Boolean,
```

Comment:
These are both essentially the same thing: `complicationConfigViewModel.loadingState`. Wouldn't it be better to pass the loading state to the Composable, and use a local variable for `loaded`? (`error` is used just once!)